torch/csrc/jit/passes/python_print.cpp

```diff
@@ -402,26 +469,33 @@ struct PythonPrintPass {
   }
 
   // map from Value to how it should be printed at each use
-  std::unordered_map<Value*, std::string> value_names_;
+  std::unordered_map<Value*, std::shared_ptr<TaggedStringStream>> value_names_;
 
-  std::string useOf(Value* v) const {
+  // NB: we MUST pass around the shared pointers to these streams by value.
+  // There is an interaction in splitLongInlines where the string value for
+  // both the RHS and the LHS of an expression are live at the same time,
+  // however the value for the RHS is overwritten in the table.
+  std::shared_ptr<TaggedStringStream> useOf(Value* v) const {
     return value_names_.at(v);
   }
-  void assignValue(Value* v, const std::string& s) {
-    value_names_[v] = s;
+  void assignValue(Value* v, std::shared_ptr<TaggedStringStream> s) {
+    value_names_[v] = std::move(s);
   }
   void assignValue(Value* v, Value* w) {
     assignValue(v, useOf(w));
   }
   void assignValuesToTheirUniqueNames(at::ArrayRef<Value*> values) {
     for (auto v : values) {
-      assignValue(v, genUniqueNameFor(v));
+      auto name_stream =
+          std::make_shared<TaggedStringStream>(&source_range_stack_);
+      (*name_stream) << genUniqueNameFor(v);
+      assignValue(v, std::move(name_stream));
```

Comment:
Example: this is assigning the wrong source to variables, because it gives them the source of their definition and not where the variable was used.